blogposts/2020/find-and-replace.md

```diff
@@ -0,0 +1,582 @@
+---
+title: "A programmer's guide to find-and-replace"
+author: Beyang Liu
+authorUrl: https://twitter.com/beyang
+publishDate: 2020-05-13T10:00-07:00
+tags: [blog]
+slug: a-programmers-guide-to-find-and-replace
+heroImage: /blog/find-replace/vscode.png
+published: true
+---
+
+So much of what we do as programmers boils down to automating the tedium out of our work. We'd like
+to spend our lives focusing on the design of beautiful abstractions and algorithms, but before we
+can get to that, we have to do the dirty work—and do it quickly.
+
+Whether you are a newbie or a seasoned engineer, chances are that you struggle with day-to-day tasks
+that seem to take longer than they should. One such task is **find and replace**.
+
+Find and replace covers a huge variety of renames, refactors, and transformations that you might
+want to apply in code. Here are a few examples:
+
+* You've made a change to a public API and need to update downstream consumers.
+* You've changed a function signature and need to update every call site.
+* You want to convert a bunch of data in one format to another (e.g., XML to JSON).
+* You're identifying and fixing common anti-patterns to improve overall code quality.
+
+Doing any of these by hand can be mind-numbing. For assistance, you can turn to a wide variety of
+find-and-replace tools. But there are so many such tools, with varying degrees of applicability,
+expressivity, ease-of-use, ease-of-learning, and scalability. How do you pick the right one?
+
+This post presents a guide to all the various find-and-replace tools I've encountered and used over
+the years, divided into three chapters:
+
+1. [Find and replace in your editor](#find-and-replace-in-your-editor)
```

Comment:
Could you represent the available tools in a matrix to show that there's a gap where campaigns fits in?

Not sure what the rows and columns would be

Reply:
Added this: 
![Selection_161](https://user-images.githubusercontent.com/1646931/82014490-92485800-9631-11ea-911b-025aca5e8b7b.png)